arrow/formatter.py

```diff
@@ -70,45 +70,45 @@ def _format_token(self, dt, token):
         if token == "ddd":
             return self.locale.day_abbreviation(dt.isoweekday())
         if token == "d":
-            return str(dt.isoweekday())
+            return f"{dt.isoweekday()}"
 
         if token == "HH":
             return f"{dt.hour:02d}"
         if token == "H":
-            return str(dt.hour)
+            return f"{dt.hour}"
         if token == "hh":
-            return "{:02d}".format(dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12))
+            return f"{dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12):02d}"
         if token == "h":
             return str(dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12))
```

Comment:
Shouldn't this be changed to an f-string as well?

Reply:
Right, fixed, repushed, thank you. Your `pre-commit` did not complain, so I ignored that line.

Reply:
That expression `dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12)` could be written as `(dt.hour - 1) % 12 + 1`, but I am not sure whether you'll like it more than the current solution.

Reply:
I usually prefer modulo to ternary statements. Any preference @systemcatch?

Reply:
I tend to prefer ternary but honestly don't mind if you guys want to change it.

Reply:
Thinking about it, I think I actually agree with Chris here. It is more explicit as it is since the hour range is obvious.

Reply:
This is actually a piece of code that never runs on my computer. A day has only 24±1 hours, not 12 :-)

Reply:
So you are saying one part of the ternary is dead code?

Reply:
No. I want to say that I personally as a user always use 24-hours format, so this piece of code formatting to 12-hours wouldn't be called on my computer. It was just a fun way to close this discussion, and it has nothing to do with this PR.

I totally accept your preference for the ternary and anyway I am happy I was able to contribute to `arrow`.

Reply:
Thanks for your contributions we really do appreciate them!